plugins/catalog-backend-module-github/src/lib/defaultTransformers.ts

```diff
@@ -0,0 +1,170 @@
+  client: typeof graphql;
+  query: string;
+  org: string;
+}
+/**
+ * Options for UserTransformer
+ *
+ * @public
+ */
+export interface UserTransformerOptions {
+  namespace?: string;
+}
+
+/**
+ * Transformer for GitHub users to UserEntity
+ *
+ * @public
+ */
+export interface UserTransformer
+  extends Transformer<User, UserEntity, UserTransformerOptions> {
+  (item: User, options: UserTransformerOptions):
+    | Promise<UserEntity>
+    | UserEntity;
+}
+
+export interface TeamTransformerOptions {
+  namespace?: string;
+}
+
+export interface TeamTransformerResponse {
+  groups: GroupEntity;
+  groupMembers: string[];
+}
+
+export interface TeamTransformer
+  extends Transformer<Team, TeamTransformerResponse, TeamTransformerOptions> {
+  (item: Team, options: TeamTransformerOptions, ctx: TransformerContext):
+    | Promise<TeamTransformerResponse>
+    | TeamTransformerResponse;
+}
+
+/**
+ * Default transformer for GitHub users to UserEntity
+ *
+ * @public
+ */
+export const defaultUserTransformer: UserTransformer = (
+  item: User,
+  options: UserTransformerOptions,
+) => {
+  const entity: UserEntity = {
+    apiVersion: 'backstage.io/v1alpha1',
+    kind: 'User',
+    metadata: {
+      name: item.login,
+      annotations: {
+        'github.com/user-login': item.login,
+      },
+    },
+    spec: {
+      profile: {},
+      memberOf: [],
+    },
+  };
+
+  if (options.namespace) entity.metadata.namespace = options.namespace;
+  if (item.bio) entity.metadata.description = item.bio;
+  if (item.name) entity.spec.profile!.displayName = item.name;
+  if (item.email) entity.spec.profile!.email = item.email;
+  if (item.avatarUrl) entity.spec.profile!.picture = item.avatarUrl;
+  return entity;
+};
+
+export const defaultOrganizationTeamTransformer: TeamTransformer = async (
+  team,
+  options,
+  ctx,
+) => {
+  const annotations: { [annotationName: string]: string } = {
+    'github.com/team-slug': team.combinedSlug,
+  };
+
+  if (team.editTeamUrl) {
+    annotations['backstage.io/edit-url'] = team.editTeamUrl;
+  }
+
+  const entity: GroupEntity = {
+    apiVersion: 'backstage.io/v1alpha1',
+    kind: 'Group',
+    metadata: {
+      name: team.slug,
+      annotations,
+    },
+    spec: {
+      type: 'team',
+      profile: {},
+      children: [],
+    },
+  };
+
+  if (options?.namespace) {
+    entity.metadata.namespace = options.namespace;
+  }
+
+  if (team.description) {
+    entity.metadata.description = team.description;
+  }
+  if (team.name) {
+    entity.spec.profile!.displayName = team.name;
+  }
+  if (team.avatarUrl) {
+    entity.spec.profile!.picture = team.avatarUrl;
+  }
+  if (team.parentTeam) {
+    entity.spec.parent = team.parentTeam.slug;
+  }
+
+  const memberNames: string[] = [];
+
+  if (!team.members.pageInfo.hasNextPage) {
+    // We got all the members in one go, run the fast path
+    for (const user of team.members.nodes) {
+      memberNames.push(user.login);
```

Comment:
yeah here is the question whether we could do `entity.spec.members!.push(` if we have that field declared on line 129 above, so we don't have to have a separate group members collection

Reply:
As above I have implemented this and seems simpler.